mysql-test/suite/versioning/t/alter.test

```diff
@@ -0,0 +1,139 @@
+--error ER_VERS_WRONG_PARAMS
+create table t(
+  a int
+) without system versioning;
+
+create table t(
+  a int
+);
+show create table t;
+--error ER_TABLE_IS_NOT_VERSIONED
+alter table t without system versioning;
+--error ER_VERS_WRONG_PARAMS
+alter table t with system versioning without system versioning;
+
+alter table t with system versioning;
+show create table t;
+
+alter table t without system versioning;
+show create table t;
+
+alter table t
+  add column trx_start timestamp(6) generated always as row start,
+  add column trx_end timestamp(6) generated always as row end,
+  add period for system_time(trx_start, trx_end),
+  with system versioning;
+show create table t;
```

Comment:
### I
Also test wrong type for `generated always` for myisam/innodb.
#### Result:
```
trx_start` must be of type `BIGINT UNSIGNED` for versioned table `#sql-6f6_3
```
#### Expected:
```
... for versioned table `t`
```
#### Implementation:
Add `original_name` to `TABLE_SHARE` and use it for `my_error` in case of `TABLE_CATEGORY_TEMPORARY`.

### II
Wrong `NOT NULL`.
#### Result:
```
  `trx_start` timestamp(6) NOT NULL GENERATED ALWAYS AS ROW START,
  `trx_end` timestamp(6) NOT NULL GENERATED ALWAYS AS ROW END,
```
#### Expected:
```
  `sys_trx_start` timestamp(6) GENERATED ALWAYS AS ROW START,
  `sys_trx_end` timestamp(6) GENERATED ALWAYS AS ROW END,
```